api/src/main/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/core/Permute.kt

```diff
@@ -0,0 +1,55 @@
+package org.jetbrains.kotlinx.dl.api.core.layer.core
+
+import org.jetbrains.kotlinx.dl.api.core.KGraph
+import org.jetbrains.kotlinx.dl.api.core.layer.Layer
+import org.jetbrains.kotlinx.dl.api.core.shape.TensorShape
+import org.tensorflow.Operand
+import org.tensorflow.Shape
+import org.tensorflow.op.Ops
+
+/**
+ * Permutes the dimensions of the input according to a given pattern.
+ * Input shape: Arbitrary
+ * Output shape: Same as input shape, but with the dimensions re-ordered according to the specified pattern.
+ * @property [dims] array of integers.
+ * @property [name] Custom layer name.
+ * @constructor Creates [Permute] object.
+ */
+public class Permute(
+    public val dims: IntArray,
+    name: String
+) : Layer(name) {
+    override fun build(tf: Ops, kGraph: KGraph, inputShape: Shape) {}
+
+    override fun computeOutputShape(inputShape: Shape): Shape {
+        val outputShape = LongArray(dims.size + 1) { 0 }
+        dims.forEachIndexed { i, dim ->
+            val target_dim = inputShape.size(dim);
+            outputShape[i + 1] = target_dim
+        }
+        return TensorShape(outputShape).toShape()
+    }
```

Comment:
The size of batch dimension is initialized as zero and remains so. To fix it, either update the value of first dimension, `outputShape[0] = inputShape.size(0)`, or just initialize `outputShape` with `inputShape` like `val outputShape = inputShape.toLongArray()`.

Further, as an alternative way, you can directly create `Shape` from a `LongArray` using: `return shapeFromDims(outputShape)`.

And a minor point: the semi-colon at the end of line 27 is redundant.